resources/assert_lib.brs

```diff
@@ -60,3 +61,96 @@ function __formatError(errMessage)
         stack: {}
     }
 end function
+
+sub __deepEquals(actual, expected, error)
+    if __roca_deepEquals(actual, expected) = true then
+        m.__pass()
+    else
+        m.__fail(m.formatError(error))
+    end if
+end sub
+
+'utility function to compare two objects/associative arrays with nested data structures
+function __roca_deepEquals(lhs as object, rhs as object) as boolean
+    ' because Brightsript does not support closures and we are doing recursion,
+    'we want to hold the function parameters in top-level state variables during each execution
+    updatedLeftState = invalid
+    updatedRightState = invalid
+    isEqual = true
+
+    ' base case
+    if type(lhs) <> type(rhs) then return false
+
+    if isArray(lhs) and isArray(rhs) then
+        ' base case
+        if lhs.count() <> rhs.count() then return false
+
+        updatedLeftState = lhs
+        updatedRightState = rhs
+
+        index = 0
+        for each updatedLeft in updatedLeftState
+            updatedRight = updatedRightState[index]
+
+            isEqual = deepEquals(updatedLeft, updatedRight)
```

Comment:
You'll probably want this to be `__roca_deepEquals` — a call back into this function using the same scope for `m`.  Looks like `deepEquals` (without the `__roca_` prefix) isn't defined in this function!